include/blast/math/StorageOrder.hpp

```diff
@@ -1,33 +1,21 @@
-// Copyright 2023 Mikhail Katliar
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+// Copyright 2023-2024 Mikhail Katliar. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
 
 #pragma once
 
-#include <blaze/math/StorageOrder.h>
-
 
 namespace blast
 {
     enum StorageOrder : bool
```

Comment:
`bool`? will there never be more `StorageOrder`s than two?

Reply:
Good point. The `StorageOrder` term was borrowed from *Blaze*, where it meant only row-major or column-major. Now I think that it could be a more complex thing, that would include panel-based and possible other storages. It could be a class that defines how to calculate memory offset given row and column indices, similar to `LayoutPolicy` in `std::mdspan`. Then we would not need special classes for panel matrices, because the `StorageOrder` parameter of the matrix class would handle accessing the elements. But making it in one step would be too big change, so I just decoupled it from *Blaze* for now.